src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -804,4 +820,195 @@ public Owner syncOwnerLastContentUpdate(Owner owner) {
+            consumer.getUuid());
+
+        List<ProductInfo> products = prodAdapter.getChildrenByProductIds(consumer.getProductIds());
+        if (products == null || products.isEmpty()) {
+            String msg = "Unable to retrieve products for anonymous cloud consumer: " + consumer.getUuid();
+            throw new RuntimeException(msg);
+        }
+
+        OffsetDateTime start = OffsetDateTime.now().minusHours(1L);
+        OffsetDateTime end = start.plusDays(2L);
+
+        CertificateSerial serial = createSerial(end);
+        KeyPair keyPair = this.pki.generateKeyPair();
+        byte[] pemEncodedKeyPair = this.pki.getPemEncoded(keyPair.getPrivate());
+
+        List<ContentInfo> contentInfo = getContentInfo(products);
+        org.candlepin.model.dto.Product container = new org.candlepin.model.dto.Product();
+        container.setContent(convertContentInfoToContentDto(contentInfo));
+        String x509Cert = createX509Cert(consumer.getUuid(), null, serial, keyPair, container,
+            BASIC_ENTITLEMENT_TYPE, start, end);
+
+        AnonymousContentAccessCertificate caCert = new AnonymousContentAccessCertificate();
+        caCert.setSerial(serial);
+        caCert.setKeyAsBytes(pemEncodedKeyPair);
+        caCert.setCert(x509Cert);
+        caCert.setContent(createAnonCloudConsumerContentPayloadAndSignature(consumer, products));
+        caCert = anonContentAccessCertCurator.create(caCert);
+
+        consumer.setContentAccessCert(caCert);
+        this.anonCloudConsumerCurator.merge(consumer);
+
+        return caCert;
+    }
+
+    private String createAnonCloudConsumerContentPayloadAndSignature(AnonymousCloudConsumer consumer,
+        Collection<ProductInfo> products) throws IOException {
+        Product engProduct = new Product()
+            .setId("content_access")
+            .setName(" Content Access");
+
+        List<ContentInfo> contentInfo = getContentInfo(products);
+        List<org.candlepin.model.Content> contents = convertContentInfoToContent(contentInfo);
+        contents.forEach(content -> engProduct.addContent(content, true));
+
+        Product skuProduct = new Product()
+            .setId("content_access")
+            .setName("Content Access");
+
+        Pool emptyPool = new Pool()
+            .setProduct(skuProduct)
+            .setStartDate(new Date())
+            .setEndDate(new Date());
+
+        Set<String> entitledProductIds = Set.of("content-access");
+
+        PromotedContent promotedContent = new PromotedContent(new AnonymousContentPrefix());
+        org.candlepin.model.dto.Product productModel = v3extensionUtil
+            .mapProduct(engProduct, skuProduct, promotedContent, null, emptyPool, entitledProductIds);
+        byte[] payloadBytes = v3extensionUtil
+            .createEntitlementDataPayload(List.of(productModel), consumer.getUuid(), emptyPool, null);
+
+        return createPayloadAndSignature(payloadBytes);
+    }
```

Comment:
@nikosmoum @Ceiu Do we even need to populate the content for the `AnonymousContentAccessCertificate` if the content paths will be read from the x509 certificate via the 1.3.6.1.4.1.2312.9.7 object ID?

Reply:
I'd need to double check to be certain, but I'm pretty sure the client uses the entitlement payload to build the .repo files, and the CDN is what checks the content paths. So we'll still need both IIRC.

Reply:
Ah yeah! I believe you are correct!